backend/src/git.rs

```diff
@@ -298,7 +309,40 @@ impl Interface {
     // }
 
     /// A code level re-implementation of `git commit`.
-    ///
+    /// A function used to checkout or create a new branch based on the name.
+    pub fn checkout_or_create_branch(repo: &Repository, branch_name: &str) -> Result<()> {
+        let head = repo.head()?;
+        let commit = head.peel_to_commit()?;
+    
+        match repo.find_branch(branch_name, BranchType::Local) {
+            Ok(_) => {
+                // If the branch exists, check it out
+                repo.set_head(&format!("refs/heads/{}", branch_name)).map_err(|e| {
+                    error!("Failed to set head to branch {branch_name}: {e:?}");
+                    e
+                })?;
+            }
+            Err(_e) => {
+                // If the branch does not exist, create it
+                match repo.branch(branch_name, &commit, false) {
+                    Ok(_) => {
+                        // Now check out the newly created branch
+                        repo.set_head(&format!("refs/heads/{}", branch_name)).map_err(|e| {
```

Comment:
I would honestly just use eyre's `.wrap_err` here, then you can just `func().wrap_err("Error!")?;